gpytorch/mlls/variational_elbo.py

```diff
@@ -30,7 +30,13 @@ def forward(self, variational_dist_f, target, **kwargs):
         log_likelihood = self.likelihood.variational_log_probability(variational_dist_f, target, **kwargs).div(
             num_batch
         )
-        kl_divergence = torch.distributions.kl.kl_divergence(variational_dist_u, prior_dist).sum().div(self.num_data)
+
+        kl_divergence = torch.distributions.kl.kl_divergence(variational_dist_u, prior_dist).unsqueeze(-1)
+
+        if log_likelihood.numel() == 1:
+            kl_divergence = kl_divergence.sum()

```

Comment:
No (e.g. the DKL CIFAR100 stuff still works) -- in the batch additive case `log_likelihood` will be a scalar so we'll be summing across all dimensions for the `kl_divergence` anyways.

Reply:
Let's say you're fitting `b` GPs, each with `l` additive terms. `log_likelihood` will be a vector of length `b`, and (I'm assuming) the kl_divergence will be a `b x l` matrix? This will fail in this case.